compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3469,7 +3455,55 @@ object Types {
     assert(resType.isInstanceOf[TermType], this)
     assert(paramNames.nonEmpty)
 
+    private def setVariances(tparams: List[LambdaParam], vs: List[Variance]): Unit =
+      if tparams.nonEmpty then
+        tparams.head.givenVariance = vs.head
+        setVariances(tparams.tail, vs.tail)
+
+    override val isVariantLambda = variances.nonEmpty
```

Comment:
I would use a more specific name highlighting that this is about lambdas with given variances.
(less important, but it would also be nicer to not overload the term "given", we could say "fixed variance" instead maybe)
```suggestion
    override val isGivenVarianceLambda = variances.nonEmpty
```

Reply:
I changed it to `isDeclaredVarianceLambda`